authors/apps/articles/tests/test_rating_article.py

```diff
@@ -0,0 +1,96 @@
+"""
+test module for rating an article
+"""
+from django.urls import reverse
+from authors.apps.articles.tests.base_class import ArticlesBaseTest
+from ..models import Article
+from .test_data import valid_article, updated_article
+
+
+class TestRatingArticle(ArticlesBaseTest):
+    """
+    test class to contain functions to test rating an article
+    """
+    def setUp(self):
+        super().setUp()
+        self.articles_url = reverse('articles:articles')
+
+    def test_rate_your_own_article(self):
+        """
+        method to test rating your own article
+        """
+        self.add_article()
+        article = Article.objects.all().first()
+        response = self.client.post(
+            reverse("articles:rating", kwargs={'slug': article.slug}),
+            data={"article": {"score": 3}},
+            format="json"
+        )
+        self.assertEqual(response.status_code, 403)
+        self.assertIn("article", response.json())
+
+    def test_rate_someones_article(self):
+        """
+        method to test rating someone's article
+        """
+        self.add_article()
+        article = Article.objects.all().first()
+        self.register_and_login_new_user()
+        response = self.client.post(
+            reverse("articles:rating", kwargs={'slug': article.slug}),
+            data={"article": {"score": 3}},
+            format="json"
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertIn("article", response.json())
+
+    def test_rate_out_of_range(self):
+        """
+        method to test rating an article using out of range values
+        """
+        self.add_article()
+        article = Article.objects.all().first()
+        self.register_and_login_new_user()
+        response = self.client.post(
+            reverse("articles:rating", kwargs={'slug': article.slug}),
+            data={"article": {"score": 6}},
+            format="json"
+        )
+        self.assertEqual(response.status_code, 400)
+        # self.assertIn("errors", response.json())
```

Comment:
Thanks for the good work @wasibani-roy, however, line 60 does not look like a comment, it looks like commented code, you could remove the block for readability.

Reply:
Change made